src/dataflow/src/render/top_k.rs

```diff
@@ -12,16 +12,21 @@ use differential_dataflow::lattice::Lattice;
 use differential_dataflow::operators::Consolidate;
 use differential_dataflow::Collection;
 use timely::dataflow::Scope;
+use timely::progress::timestamp::Refines;
+use timely::progress::Timestamp;
 
 use expr::MirRelationExpr;
 use repr::{Diff, Row};
 
 use crate::render::context::Context;
+use crate::render::SystemEventTime;
 
 // The implementation requires integer timestamps to be able to delay feedback for monotonic inputs.
-impl<G> Context<G, MirRelationExpr, Row, repr::Timestamp>
+impl<G, T> Context<G, MirRelationExpr, Row, T>
 where
-    G: Scope<Timestamp = repr::Timestamp>,
+    G: Scope,
+    G::Timestamp: Lattice + Refines<T> + SystemEventTime,
```

Comment:
its not obvious how these stated bounds differ from things like:

```
    G::Timestamp: Lattice,
    T: Refines<G::Timestamp> + Lattice + SystemEventTime,
```

I confirmed that those bounds don't just work if I try copy pasting them in, but its a bit mysterious as to why (I _think_ it has to do with the use of `Variable`s?)